libs/service/api_service.cpp

```diff
@@ -80,11 +64,40 @@ _build_helper_id_to_address_map(
         return win32_error_code_to_ebpf_result(result);
     }
 
-    helper_addresses.resize(helper_id_mapping.size());
-
     index = 0;
-    for (auto& address : helper_addresses) {
-        address = reply->address[index++];
+    for (auto& [helper_id, helper_address] : helper_id_to_address) {
+        helper_address = reply->address[index++];
+    }
+
+    return EBPF_SUCCESS;
+}
+
+static ebpf_result_t
+_build_helper_id_to_address_map(
+    _In_reads_(instruction_count) ebpf_inst* instructions,
+    uint32_t instruction_count,
+    const std::map<uint32_t, uint64_t>& helper_id_to_address,
+    uint32_t& unwind_index)
+{
+    // Note:
+    // eBPF supports helper IDs in the range [1, MAXUINT32]
+    // uBPF jitter only supports helper IDs in the range [0,63]
+    // Build a table to map [1, MAXUINT32] -> [0,63]
+    std::map<uint32_t, uint32_t> helper_id_mapping;
+    unwind_index = MAXUINT32;
+
+    uint32_t index = 0;
+    for (auto [helper_id, helper_address] : helper_id_to_address) {
+        helper_id_mapping[helper_id] = index++;
+    }
+
+    if (helper_id_mapping.size() == 0) {
+        return EBPF_SUCCESS;
+    }
+
+    // uBPF jitter supports a maximum of 64 helper functions
+    if (helper_id_mapping.size() > 64) {
```

Comment:
Can we check helper_id_to_address.size() as the first check and return error if > 64, as opposed to populating the map only to see it fail?

Reply:
Done